packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -235,7 +313,7 @@ library LibProposing {
 
         // Hash the assignment with the blobHash, this hash will be signed by
         // the prover, therefore, we add a string as a prefix.
-        bytes32 hash = hashAssignment(assignment, blobHash);
+        bytes32 hash = hashAssignment(assignment, address(this), blobHash);
```

Comment:
Good point. This may also apply to other sendEther case.

Reply:
After a second thought: if the prover can change the gas used when receive Ether, then the proposer should be carefully and should probably not choose to use the prover. In this PR, https://github.com/taikoxyz/taiko-mono/pull/15200, I applied a max gas, but the value is set to 200K to prevent the worst case.